src/main/scala/beam/agentsim/agents/modalbehaviors/DrivesVehicle.scala

```diff
@@ -323,21 +322,15 @@ trait DrivesVehicle[T <: DrivingData] extends BeamAgent[T] with HasServices {
       // Produce link events for this trip (the same ones as in PathTraversalEvent).
       // TODO: They don't contain correct timestamps yet, but they all happen at the end of the trip!!
       // So far, we only throw them for ExperiencedPlans, which don't need timestamps.
+      val head = data.passengerSchedule.schedule
+        .drop(data.currentLegPassengerScheduleIndex).head
```

Comment:
Unless I missed an area, you could even just get the first part of the tuple here and use it...as the second part is never used anyway.